### PR TITLE
Part 3 - Convert action tests to mocha

### PR DESCRIPTION
**Pull request by linclark**
This depends on existing PRs. Will rebase once #159 has been committed

### CODE REVIEW

**Comment by linclark** [comment]
@bgrins this is rebased and ready for review

File: devtools/client/webconsole/new-console-output/test/actions/messages.test.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+      };
+      const action = messageAdd(packet);
+      const expected = {
+        type: constants.MESSAGE_ADD,
+        message: stubConsoleMessages.get("console.log('foobar', 'test')")
+      };
+
+      // Some values on the message are generated by prepareMessage. Manually set
+      // these on the expected message to match.
+      expected.message = expected.message.set("repeatId", getRepeatId(expected.message));
```
Comment (bgrins) [inline devtools/client/webconsole/new-console-output/test/actions/messages.test.js:49]: This looks like a slight change in behavior with the xpc shell test.  It used to (1) not touch repeatId but would instead call prepareMessage and (2) would remove the 'id' property from both action and expected.  What's the reasoning for the change?

**Comment by bgrins** [comment]
Looks good overall, just one comment

File: devtools/client/webconsole/new-console-output/test/actions/messages.test.js (hunk 1 above)
Comment (linclark) [inline devtools/client/webconsole/new-console-output/test/actions/messages.test.js:49]: Eventually once the stubs are autogenerated in #169, they will have the expected repeatIds (and an id of 1), so generating it won't be necessary.

**Pull request merged**